### PR TITLE
perf: reuse splines on UVBeam objects

### PR DESCRIPTION
**Pull request by steven-murray**
Re-uses splines when using `beam_list` instead of `bm_cube`. This should give a healthy speedup.

### CODE REVIEW

**Review by piyanatk — APPROVED** [review]
All good @steven-murray. I approved.

File: src/vis_cpu/conversions.py [hunk 1]
```diff
@@ -392,3 +392,34 @@ def uvbeam_to_lm(
...
+    zenith (at 1e-18) to ensure that we can "interpolate away" from zenith properly.
+
+    Since we interpolate in real/imag, the phase flip at zenith will always mean that
+    there is a point with magnitude of zero between zenith and one of the points on the
+    circle closest to zenith. By making this circle very small (1e-18) we reduce the
+    chance that this artifact has any effect.
+    """
+    if not bm_pix % 2:
+        raise ValueError("Must use odd bm_pix so that you sample zenith!")
+
```
Comment (piyanatk) [inline src/vis_cpu/conversions.py:412]: Should we also enforce the minimum values of bm_pix? It is kinda unnecessary but we have to make sure that at least `bm_pix // 2 -1` in line 420 doesn't go negative

File: src/vis_cpu/conversions.py [hunk 2]
```diff
@@ -392,3 +392,34 @@ def uvbeam_to_lm(
...
+    # The smallest value of lm should always be zero, however numerically it doesn't
+    # always quite work out, so we force it to be exactly zero.
+    lm[np.argmin(np.abs(lm))] = 0.0
+    lm = np.insert(
+        lm,
+        bm_pix // 2 - 1,
+        -1e-18,
+    )
+    lm = np.insert(lm, bm_pix // 2 + 1, 1e-18)
```
Comment (r-pascua) [inline src/vis_cpu/conversions.py:423]: Can you add a unit test to make sure this does the right thing? It looks like it should, but I think it's good to write a small test for it.

**Review by r-pascua — APPROVED** [review]
This looks good to me, though I would like to see the test I request. You mentioned that this now depends on a `pyuvdata` PR that hasn't been merged yet, but I don't quite see how. Could you explain? Is it the `check_azza_domain` kwarg?

**Review by piyanatk — APPROVED** [review]
Look good @steven-murray . Just one comment that might be unnecessary. I have approved.

File: src/vis_cpu/conversions.py (hunk 2 above)
Comment (steven-murray) [reply src/vis_cpu/conversions.py:423]: Yup, good idea.

File: src/vis_cpu/conversions.py (hunk 1 above)
Comment (steven-murray) [reply src/vis_cpu/conversions.py:412]: Yeah, I can do that :-)

**Comment by steven-murray** [comment]
Thanks @r-pascua and @piyanatk -- I've made the relevant changes.

To your question, @r-pascua, yes the update is the `check_azza_domain` parameter, which will become available in pyuvdata soon.

**Pull request merged**